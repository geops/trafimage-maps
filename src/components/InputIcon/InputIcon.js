import React from "react";
import PropTypes from "prop-types";
import { makeStyles, useTheme } from "@mui/styles";

const useStyles = makeStyles((theme) => {
  return {
    box: {
      position: "relative",
      width: 16,
      height: 16,
      border: `1px solid lightgray`,
      display: "flex",
      alignItems: "center",
      justifyContent: "center",
      zIndex: 2,
      "&:hover": {
        backgroundColor: "rgba(0, 0, 0, 0.1)",
      },
      "& svg": {
        margin: 2,
      },
    },
    radio: {
      borderRadius: "50%",
    },
    innerCircle: {
      position: "absolute",
      borderRadius: "50%",
      width: 12,
      height: 12,
      backgroundColor: theme.palette.secondary.main,
    },
  };
});

function InputIcon({ checked, type, "data-cy": dataCy }) {
  const classes = useStyles();
  const theme = useTheme();
  return (
    <div
      className={`${classes.box}${type === "radio" ? ` ${classes.radio}` : ""}`}
      data-cy={dataCy}
    >
      {checked && type === "checkbox" && (
        <svg
          width="13.707"
          height="10.051"
          viewBox="0 0 3.627 2.659"
          data-testid="input-icon-tick"
        >
          <path
            fill={theme.palette.secondary.main}
            fillRule="evenodd"
            d="m43.353 31.363-9 8.99-.353.354-.353-.354-4-4 .707-.707L34 39.293l8.647-8.637z"
            clipRule="evenodd"
            transform="matrix(.26458 0 0 .26458 -7.844 -8.111)"
          />
        </svg>
      )}
      {checked && type === "radio" && (
        <div className={classes.innerCircle} data-testid="input-icon-radio" />
      )}
    </div>
  );
}

InputIcon.propTypes = {
  checked: PropTypes.bool,
  type: PropTypes.oneOf(["checkbox", "radio"]),
  "data-cy": PropTypes.string,
};
InputIcon.defaultProps = {
  checked: false,
  type: "checkbox",
  "data-cy": undefined,
};

export default InputIcon;
