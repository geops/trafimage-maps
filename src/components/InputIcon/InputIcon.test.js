import React from "react";
import { render } from "@testing-library/react";
import { ThemeProvider } from "@mui/material";
import theme from "../../themes/default";
import InputIcon from "./InputIcon";

describe("InputIcon", () => {
  test("should match snapshot when unchecked and checkbox", () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <InputIcon />
      </ThemeProvider>,
    );
    expect(container.innerHTML).toMatchSnapshot();
  });
  test("should match snapshot when unchecked and radio button", () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <InputIcon type="radio" />
      </ThemeProvider>,
    );
    expect(container.innerHTML).toMatchSnapshot();
  });
  test("should match snapshot when checked and checkbox", () => {
    const { container, getByTestId } = render(
      <ThemeProvider theme={theme}>
        <InputIcon checked />
      </ThemeProvider>,
    );
    expect(getByTestId("input-icon-tick")).toBeInTheDocument();
    expect(container.innerHTML).toMatchSnapshot();
  });
  test("should match snapshot when checked and radio button", () => {
    const { container, getByTestId } = render(
      <ThemeProvider theme={theme}>
        <InputIcon type="radio" checked />
      </ThemeProvider>,
    );
    expect(getByTestId("input-icon-radio")).toBeInTheDocument();
    expect(container.innerHTML).toMatchSnapshot();
  });
});
