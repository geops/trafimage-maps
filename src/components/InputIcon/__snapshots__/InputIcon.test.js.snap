// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputIcon should match snapshot when checked and checkbox 1`] = `
<div class="makeStyles-box-7">
  <svg width="13.707"
       height="10.051"
       viewbox="0 0 3.627 2.659"
       data-testid="input-icon-tick"
  >
    <path fill="#eb0000"
          fill-rule="evenodd"
          d="m43.353 31.363-9 8.99-.353.354-.353-.354-4-4 .707-.707L34 39.293l8.647-8.637z"
          clip-rule="evenodd"
          transform="matrix(.26458 0 0 .26458 -7.844 -8.111)"
    >
    </path>
  </svg>
</div>
`;

exports[`InputIcon should match snapshot when checked and radio button 1`] = `
<div class="makeStyles-box-10 makeStyles-radio-11">
  <div class="makeStyles-innerCircle-12"
       data-testid="input-icon-radio"
  >
  </div>
</div>
`;

exports[`InputIcon should match snapshot when unchecked and checkbox 1`] = `
<div class="makeStyles-box-1">
</div>
`;

exports[`InputIcon should match snapshot when unchecked and radio button 1`] = `
<div class="makeStyles-box-4 makeStyles-radio-5">
</div>
`;
